npep/npep-137-conformance-profiles.md

```diff
@@ -0,0 +1,187 @@
+can leverage when running the conformance test suite in their downstream repos
+to report the results of the tests back to the Network Policy API project and
+receive recognition (eg. "conformance badge")
+
+## Goals
+
+* Add conformance profiles for existing tests that downstream implementations
+can subscribe to to run tests associated to the supported API feature sets
+
+* Add a reporting mechanism via a new CRD where conformance results can be
+reported back to the Network Policy API project and provide "conformance badges"
+via Network Policy official website to recognize these implementations.
+
+* Expand existing conformance testing framework to account for profiles
+
+## Non-Goals
+
+* The first iteration will just have a simple repo under which the report
+results will be stored, no automated reporting infrastructure will be built for this now.
+Implementations will need to open PRs against network-policy-api repo to upload
+the results.
+
+## Introduction
+
+Currently, the conformance tests are grouped into `StandardCoreFeatures` and
+`StandardExtendedFeatures`. The features that fall under the `StandardCoreFeatures`
+are a must for the implementations to be conformant while the features that fall
+under `StandardExtendedFeatures` are opt-ins which are not mandatory.
+
+In this NPEP, we will add a concept called `named profiles` which will indicate a
+"level of conformance" on a resource level. We are picking profiles on a resource
+level because that is what makes sense for the project as of today. This can be
+changed as the project evolves. Using these `named profiles`, implementations can
+run the conformance tests and prove they satisfy the requirements for each profile
+and receive conformance badges from the Network Policy API project.
+
+This NPEP aims to provide an API (CRD) for reporting results from these conformance
+tests. CLI tooling will also be provided for invoking the conformance tests with
+the profile names so that the results can be easily submitted back upstream.
+
+## User-Stories/Use-Cases
+
+Story 1: Receive an API conformance badge
+
+As a CNI maintainer, I want to receive recognition from Network Policy API repo
+that my plugin implements `AdminNetworkPolicy` API. This badge will enable us
+to advertise upstream feature support in our plugin easily to our users.
+
+Story 2: Track Implementations for each API resource
+
+As a Network Policy project maintainer, I would like to easily track which
+implementations are using the defined API resources in our project so that it
+paves way for easier collaboration and feedback loops when changes are introduced
+to the API in each release.
+
+## API
+
+We will add a new API resource called `ConformanceReport` which will be at
+the center of our test result reporting workflow. The implementors running
+the tests can then:
+
+1. Choose a `named profile`
+2. Add integration for the tests under this profile in their downstream project
```

Comment:
```suggestion
2. Integrate the tests required by a given profile in their downstream project
```
Is this what you were trying to say?

Reply:
yes :D changed it according to your words which make it more clear.